lib/iris/analysis/__init__.py

```diff
@@ -2316,13 +2316,17 @@ def __repr__(self):
 
 def clear_phenomenon_identity(cube):
     """
-    Helper function to clear the standard_name, attributes, and
```

Comment:
This isn't a good extension of the function, because cell-measures and ancils are *not* part of phenomenon identity.
This is also a public API function, so it does need to do what it says on the tin.

I reckon the additional aspects should be separately handled, adjacent to where [this is function is called in `_math_op_common`](https://github.com/SciTools/iris/blob/b0027d4b72e7dd43e07a6d6571d93d23f4d8c641/lib/iris/analysis/maths.py#L946).